### PR TITLE
Setup/vargrant

### PR DESCRIPTION
**Pull request by fsherratt**

### CODE REVIEW

**Comment by fsherratt** [comment]
Current state:
- It works on multiple platforms — I have successfully tested it on my Intel Mac (Quad-Core), Intel Laptop (Dual-Core) and AMD Desktop (Hexa-Core)
- Takes between 30mins to 2hr30 to fully install

Things to improve:
- I don't like the fact we're not specifying the exact versions of everything
  - PX4 is currently cloning directly from master - `ubuntu.sh` script in the latest stable release doesn't work
  - Need a requirements.txt file for pip3 install
  - Many other packages are just downloading the latest
- I think it would be better to build our own base Ubuntu vagrant box ([Vagrant — Creating a Base Box](https://www.vagrantup.com/docs/providers/virtualbox/boxes))
  - I've had issues with out of date kernels
  - Installing the GUI is not ideal
  - We could pre-install things we are always going to want
  - The graphics system is a little dodgy — resizing of the window doesn't always work
- PX4 Gazebo ROS launcher doesn't work with 3D graphics acceleration enable — this limits FPS

File: bootstrap.sh [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+# Add ROS source to bashrc
+#NOTE THIS ONLY WORKS FOR NOETIC
+# If we cannot find the text "noetic" in our default bashrc, add it.
+if ! grep -q noetic $USER_HOME/.bashrc 
+then
+	echo "Adding ros noetic to "$USER" .bashrc..."
+	echo "source /opt/ros/noetic/setup.bash" >> $USER_HOME/.bashrc 
+	source $USER_HOME/.bashrc
+
+elsecd /home
```
Comment (Neoxas) [inline bootstrap.sh]: Will this else code work in the ROS?

**Review by Neoxas — CHANGES_REQUESTED** [review]
Fix else plz

File: bootstrap.sh (hunk 1 above)
Comment (fsherratt) [reply bootstrap.sh]: Fixed in 2cbf005

**Review by Neoxas — APPROVED** [review]

**Pull request merged**